### PR TITLE
feat: fetch and save GSTIN details to validate Status in Transactions

### PR DESCRIPTION
**Pull request by vishakhdesai**
New Doctype: GSTIN Detail
Fields: 
1. GSTIN
2. Status
3. Registration Date
4. Cancelled Date
5. Last Updated On

Update or create the GSTIN whenever the get_gstin_info api is called.
Show status of GSTIN on customer and supplier details.

Validate GSTIN status on following doctypes:

1. Purchase Invoice
2. Purchase Receipt
3. Purchase Order
4. Quotation
5. Sales Order
6. Delivery Note
7. Sales Invoice
8. POS Invoice

### CODE REVIEW

File: india_compliance/gst_india/client_scripts/party.js [hunk 1]
```diff
@@ -115,14 +115,32 @@ function show_overseas_disabled_warning(doctype) {
...
+            let data = await india_compliance.get_gstin_options(frm.doc.name, doctype);
+            field.set_data(data);
+
+            frappe.call({
+                method: "india_compliance.gst_india.doctype.gstin_detail.gstin_detail.get_gstin_status",
+                args: {
+                    gstin: data[0]
+                },
+                callback: (res) => {
+                    field.set_description(
```
Comment (vorasmit) [inline india_compliance/gst_india/client_scripts/party.js]: add this to public js.

File: india_compliance/gst_india/doctype/gstin_detail/gstin_detail.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+   "fieldtype": "Datetime",
+   "label": "Last Updated On"
+  }
+ ],
+ "index_web_pages_for_search": 1,
+ "links": [],
+ "modified": "2023-06-19 18:16:47.324987",
+ "modified_by": "Administrator",
+ "module": "GST India",
+ "name": "GSTIN Detail",
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstin_detail/gstin_detail.json]: Let's just call this GSTIN

File: india_compliance/gst_india/utils/gstin_info.py [hunk 3]
```diff
@@ -56,6 +57,26 @@ def get_gstin_info(gstin):
...
 
 
+def update_gstin_details(response):
+    registration_date = response.rgdt
+    cancelled_date = response.cxdt
+
+    try:
+        cancelled_date = parse_datetime(cancelled_date, day_first=True)
+    except Exception:
+        cancelled_date = None
```
Comment (vorasmit) [inline india_compliance/gst_india/utils/gstin_info.py]: Better with if condition.

If cancelled_date: ...

Comment (vishakhdesai) [reply india_compliance/gst_india/utils/gstin_info.py]: But sometimes the cancelled_date have value like : "-"

File: india_compliance/gst_india/doctype/gst_settings/gst_settings.json [hunk 4]
```diff
@@ -78,7 +79,7 @@
   },
   {
    "default": "1",
-   "description": "As per <a href=\"https://old.cbic.gov.in/resources/htdocs-cbec/gst/notfctn-78-central-tax-english-2020.pdf\">Central Tax Notification No. 78/2020</a>, HSN/SAC code must be specified in Sales Invoice with atleast 4 or 6 digits (based on turnover)",
+   "description": "As per <a href=\"https://www.cbic.gov.in/resources/htdocs-cbec/gst/notfctn-78-central-tax-english-2020.pdf\">Central Tax Notification No. 78/2020</a>, HSN/SAC code must be specified in Sales Invoice with atleast 4 or 6 digits (based on turnover)",
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gst_settings/gst_settings.json]: .

File: india_compliance/gst_india/client_scripts/party.js [hunk 5]
```diff
@@ -122,7 +122,21 @@ function set_gstin_query(doctype) {
             frm._gstin_options_set_for = frm.doc.name;
             const field = frm.get_field("gstin");
             field.df.ignore_validation = true;
-            field.set_data(await india_compliance.get_gstin_options(frm.doc.name, doctype));
+            let data = await india_compliance.get_gstin_options(frm.doc.name, doctype);
+            field.set_data(data);
+
+            frappe.call({
```
Comment (vorasmit) [inline india_compliance/gst_india/client_scripts/party.js]: Address.js

**Pull request merged**